compiler/src/test/java/com/squareup/anvil/compiler/dagger/AssistedFactoryGeneratorTest.kt

```diff
@@ -1968,4 +1969,25 @@ public final class AssistedServiceFactory_Impl implements AssistedServiceFactory
       .compile(*sources)
       .apply(block)
   }
+
+  /**
+   * Removes parameters of the functions in a String like
+   * ```
+   * [create([string]), get([])]
+   * ```
+   * That's necessary, because the output of the parameters is slightly different between Anvil
+   * and Dagger. Dagger also doesn't guarantee any order of functions.
+   */
+  private fun String.removeParameters(): String {
+    val start = 1 + (indexOf('[').takeIf { it >= 0 } ?: return this)
+    val end = indexOfLast { it == ']' }.takeIf { it >= 0 } ?: return this
```

Comment:
```suggestion
    val end = indexOf(']').takeIf { it >= 0 } ?: return this
```

?

Reply:
Oh whoops, `indexOfLast`. Never mind.

Reply:
Notice the sample string: `[create([string]), get([])]`. It has to be the index of the last match.